rosbag2_compression/test/rosbag2_compression/test_sequential_compression_writer.cpp

```diff
@@ -53,10 +57,29 @@ class SequentialCompressionWriterTest : public Test
   std::shared_ptr<StrictMock<MockConverterFactory>> converter_factory_;
   std::unique_ptr<MockMetadataIo> metadata_io_;
   std::unique_ptr<rosbag2_cpp::Writer> writer_;
-  rosbag2_cpp::StorageOptions storage_options_;
+  rcpputils::fs::path tmp_dir_;
+  rosbag2_cpp::StorageOptions tmp_dir_storage_options_;
   std::string serialization_format_;
 };
 
+TEST_F(SequentialCompressionWriterTest, open_throws_on_empty_storage_options_uri)
+{
+  rosbag2_compression::CompressionOptions compression_options{
+    "zstd", rosbag2_compression::CompressionMode::FILE};
+  auto compression_factory = std::make_unique<rosbag2_compression::CompressionFactory>();
+
+  auto sequential_writer = std::make_unique<rosbag2_compression::SequentialCompressionWriter>(
+    compression_options,
+    std::move(compression_factory),
+    std::move(storage_factory_),
+    converter_factory_,
+    std::move(metadata_io_));
+  writer_ = std::make_unique<rosbag2_cpp::Writer>(std::move(sequential_writer));
+
+  EXPECT_THROW(
```

Comment:
http://build.ros2.org/job/Rpr__rosbag2__ubuntu_focal_amd64/209/testReport/rosbag2_compression/SequentialCompressionWriterTest/open_throws_on_empty_storage_options_uri/, currently this is expected to fail unless https://github.com/ros2/rcpputils/pull/95 is merged.